node/core/dispute-coordinator/src/real/mod.rs

```diff
@@ -648,21 +861,36 @@ async fn handle_import_statements(
 
 			return Ok(ImportStatementsResult::InvalidImport)
 		},
-		Some(info) => info.validators.clone(),
+		Some(info) => info,
 	};
+	let validators = session_info.validators.clone();
 
 	let n_validators = validators.len();
 
 	let supermajority_threshold = polkadot_primitives::v1::supermajority_threshold(n_validators);
 
-	let mut votes = overlay_db
+	// In case we are not provided with a candidate receipt
+	// we operate under the assumption, that a previous vote
+	// which included a `CandidateReceipt` was seen.
+	// This holds since every block is preceeded by the `Backing`-phase.
```

Comment:
Unfortunately not, but it's _really, really unlikely_ that we haven't seen the candidate receipt. It would require a pretty complex and fairly arbitrary block withholding attack that serves no purpose because this branch is only hit when disputes are already concluded and therefore already have 34% honest participation.

Reply:
I'll extend your comment, and it's dealt with by a graceful early return

Reply:
When we revert a dispute we also typically revert to a point after it was already backed so unless an attacker is chaining reverts it's almost certain that the validators have all seen the candidate receipt.